packages/path_provider/path_provider_android/lib/path_provider_android.dart

```diff
@@ -81,7 +81,12 @@ class PathProviderAndroid extends PathProviderPlatform {
   }
 
   @override
-  Future<String?> getDownloadsPath() {
-    throw UnsupportedError('getDownloadsPath is not supported on Android');
+  Future<String?> getDownloadsPath() async {
+    final List<String>? paths =
+        await getExternalStoragePaths(type: StorageDirectory.downloads);
+    if (paths != null && paths.isNotEmpty) {
+      return paths.first;
+    }
+    return null;
   }
```

Comment:
this one I've updated to catch the stateError scenario when externalStoragePaths returns empty list.